apps/core/templatetags/core_tags.py

```diff
@@ -20,3 +20,41 @@ def navigation(context):
         "current_page": context.get("page"),
         "annotated_list": Page.get_annotated_list_qs(pages),
     }
+
+
+@register.filter
+def next_page(value):
+    page = value
+    if page.get_descendants():
+        return page.get_descendants().live().first()
+    elif page.get_siblings().live().filter(path__gt=page.path):
+        return page.get_siblings().live().filter(path__gt=page.path).first()
+    elif page.get_ancestors().live().last().get_siblings().filter(path__gt=page.path):
+        return (
+            page.get_ancestors()
+            .live()
+            .last()
+            .get_siblings()
+            .filter(path__gt=page.path)
+            .first()
+        )
+    else:
+        return None
```

Comment:
Note that each `get_descendants()` and `get_siblings()` call returns a QuerySet, and using it directly in an `if` expression would evaluate the queryset (i.e. execute the query), returning all objects that match the query. This means that there would be some duplicated queries.

Possible alternative:

```suggestion
@register.filter
def next_page(page):
    first_descendant = page.get_descendants().live().first()
    if first_descendant:
        return first_descendant

    next_sibling = page.get_siblings().live().filter(path__gt=page.path).first()
    if next_sibling:
        return next_sibling

    next_section = (
        page.get_ancestors()
        .live()
        .last()
        .get_siblings()
        .filter(path__gt=page.path)
        .first()
    )
    if next_section:
        return next_section

    return None
```

Reply:
Oh I see... Thanks for the suggestion, I will update the PR.